lib/Dashboard/ActivityWidget.php

```diff
@@ -83,6 +118,49 @@ public function getUrl(): ?string {
 	 * @inheritDoc
 	 */
 	public function load(): void {
-		\OCP\Util::addScript('activity', 'activity-dashboard');
+		Util::addScript('activity', 'activity-dashboard');
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	public function getItems(string $userId, ?string $since = null, int $limit = 7): array {
+		$activities = $this->data->get(
+			$this->helper,
+			$this->settings,
+			$userId,
+			$since ? (int) $since : 0,
+			50,
+			'desc',
+			'by',
+			'',
+			0
+		);
+		return array_map(function (array $activity) {
+			return new WidgetItem(
+				$activity['subject'],
+				$this->dateTimeFormatter->formatTimeSpan($activity['timestamp']),
+				$activity['link'],
+				$this->urlGenerator->getAbsoluteURL(
+					$this->urlGenerator->linkToRoute('core.avatar.getAvatar', ['userId' => $activity['affecteduser'], 'size' => 44])
```

Comment:
affecteduser is always "You". I don't think that is the icon we should show. Should use the `$event->getIcon()` instead?

Reply:
Yeah I meant to use the `user` prop just like the frontend widget does. Anyway, you're right, the `icon` is better (but not consistent with the "javascript" widget).

Reply:
The webfrontend shows the avatar + activity icon
https://github.com/nextcloud/activity/pull/920

Reply:
Ok then the action icon might be better than the user avatar. Done.